js/modules/k6/html/html.go

```diff
@@ -44,17 +47,201 @@ type Selection struct {
 	sel *goquery.Selection
 }
 
-func (s Selection) Add(arg goja.Value) Selection {
-	switch val := arg.Export().(type) {
+func (s Selection) emptySelection() Selection {
+	// Ask for out of bounds item for an empty selection.
+	return s.Eq(s.Size())
+}
+
+func (s Selection) buildMatcher(v goja.Value, gojaFn goja.Callable) func(int, *goquery.Selection) bool {
+	return func(idx int, sel *goquery.Selection) bool {
+		fnRes, fnErr := gojaFn(v, s.rt.ToValue(idx), s.rt.ToValue(sel))
+
+		if fnErr != nil {
+			panic(fnErr)
+		}
+
+		return fnRes.ToBoolean()
+	}
+}
+
+func (s Selection) varargFnCall(arg interface{},
```

Comment:
Ah, that's neater.

Um, that smiley in my comment was meant to be :\\. But the slash was removed and it was in brackets so now it looks glib and like I was happy but I wasn't.